frontend/device/remarkable/device.lua

```diff
@@ -17,39 +29,61 @@ local Remarkable = Generic:new{
     display_dpi = 226,
     -- Despite the SoC supporting it, it's finicky in practice (#6772)
     canHWInvert = no,
-    home_dir = "/mnt/root",
+    home_dir = "/home/root",
 }
 
-local EV_ABS = 3
-local ABS_X = 00
-local ABS_Y = 01
-local ABS_MT_POSITION_X = 53
-local ABS_MT_POSITION_Y = 54
--- Resolutions from libremarkable src/framebuffer/common.rs
-local screen_width = 1404 -- unscaled_size_check: ignore
-local screen_height = 1872 -- unscaled_size_check: ignore
-local wacom_width = 15725 -- unscaled_size_check: ignore
-local wacom_height = 20967 -- unscaled_size_check: ignore
-local wacom_scale_x = screen_width / wacom_width
-local wacom_scale_y = screen_height / wacom_height
-local mt_width = 767 -- unscaled_size_check: ignore
-local mt_height = 1023 -- unscaled_size_check: ignore
-local mt_scale_x = screen_width / mt_width
-local mt_scale_y = screen_height / mt_height
-local adjustTouchEvt = function(self, ev)
+local Remarkable1 = Remarkable:new{
+    model = "reMarkable",
+    mt_width = 767, -- unscaled_size_check: ignore
+    mt_height = 1023, -- unscaled_size_check: ignore
+    input_wacom = "/dev/input/event0",
+    input_ts = "/dev/input/event1",
+    input_buttons = "/dev/input/event2",
+    battery_path = "/sys/class/power_supply/bq27441-0/capacity",
+    status_path = "/sys/class/power_supply/bq27441-0/status",
+}
+
+function Remarkable1:adjustTouchEvent(ev, by)
+    if ev.type == EV_ABS then
+        ev.time = TimeVal:now()
+        -- Mirror X and Y and scale up both X & Y as touch input is different res from
+        -- display
+        if ev.code == ABS_MT_POSITION_X then
+            ev.value = (Remarkable1.mt_width - ev.value) *  by.mt_scale_x
```

Comment:
`ev.value = (self.mt_width - ev.value) * by.mt_scale_x`

Reply:
it crashes with nil exception

Reply:
it should / could be `self.device.mt_width`

Reply:
yup, because `self` is Input, right? :smile: 

Feel free to do want you want. If you go the `self.device.mt_width` route we can have a single adjustHook. Just use `touch_x_mirrored` or something similar as part of device definition and query its value with `self.device.touch_x_mirrored`. AFAICT that's the only different in the implementation between these two devices